### PR TITLE
Platform Services Docs

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description

- Add latest examples in Overview page
- Remove references to individual services pages. Link to the docs in
  repositories instead.
- Add an example for overriding the network service.

## Related Issue
#146 #143 

## How Has This Been Tested?
- Ran `yarn dev` locally to verify formatting.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/documentation/mobile-core/platform-services/tabs/network-service.md [hunk 1]
```diff
@@ -29,4 +48,229 @@ let networkService = ServiceProvider.shared.networkService
...
+                // handle exceptions
+            }
+
+            connection.disconnect();
+        }
+    }
+}
+```
+
+2. Create a custom implementation of `Networking` interface and implement the `connectAsync` method. The implementation of `connectAsync` should handle the connection establishment with the details provided in the `NetworkRequest` and, notify the caller of a response using the `NetworkCallback` parameter.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/platform-services/tabs/network-service.md:130]: "... implementation of the `Networking` interface and..."

File: src/pages/documentation/mobile-core/platform-services/tabs/network-service.md [hunk 2]
```diff
@@ -29,4 +48,229 @@ let networkService = ServiceProvider.shared.networkService
...
+}
+```
+
+Once you have implemented the `Networking` protocol, all that's left is to override the ServiceProvider's NetworkService as follows:
+
+```swift
+ServiceProvider.shared.networkService = SampleNetworkOverride()
+```
+
+If you would like to revert back to the default `NetworkService`, simply set the `networkService` to nil.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/platform-services/tabs/network-service.md:272]: We can just make this "... set the `networkService` to nil.".

**Review by calebk1 — COMMENTED** [review]
Looks good! Just a few comments.

File: src/pages/documentation/mobile-core/platform-services/index.md [hunk 3]
```diff
@@ -2,23 +2,15 @@ import Tabs from './tabs/index.md'
...
-<InlineAlert variant="info" slots="text"/>
-
-This feature is only available in [Android Core 1.8.0](../../release-notes/2021.md#android-core-1-8-0) or later and [iOS Core 3.0.0](../../release-notes/2021.md#adobe-experience-platform-ios-core-sdks) or later.
-
-| Service  | Android SDK | iOS SDK |
-| :------- | :---------- | :------ |
-| [Networking](./network-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
-| [DataQueue](./data-queue-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
-| [SystemInfoService/DeviceInforming](./system-device-info-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
+The Platform Services are provided by the Adobe Experience Platform Mobile SDKs as part of the Mobile Core extension. These services provide shared functionality throughout the SDK that can be shared by extensions. For example, services provide shared functionality for networking, data queuing, caching, and more. For more information on services provided by the SDK please see documentation in our [iOS](https://github.com/adobe/aepsdk-core-ios/blob/main/Documentation/Services/README.md) and [Android](https://github.com/adobe/aepsdk-core-ios/tree/main/Documentation/Services) repositories.
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply src/pages/documentation/mobile-core/platform-services/index.md:5]: Most of the services provided have auxiliary components that need explanation. Explaining them without adding detailed code and pointers to the classes is difficult. Also, most service implementations are not frequently overridden by customers. So we decided to redirect implementers to docs in the repos instead. 
Given that  some customers override network service we agreed to keep it here for now.

File: src/pages/documentation/mobile-core/platform-services/index.md [hunk 4]
```diff
@@ -2,23 +2,15 @@ import Tabs from './tabs/index.md'
...
-<InlineAlert variant="info" slots="text"/>
-
-This feature is only available in [Android Core 1.8.0](../../release-notes/2021.md#android-core-1-8-0) or later and [iOS Core 3.0.0](../../release-notes/2021.md#adobe-experience-platform-ios-core-sdks) or later.
-
-| Service  | Android SDK | iOS SDK |
-| :------- | :---------- | :------ |
-| [Networking](./network-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
-| [DataQueue](./data-queue-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
-| [SystemInfoService/DeviceInforming](./system-device-info-service.md) | Core 1.8.0 + | AEPCore 3.0.0 + |
+The Platform Services are provided by the Adobe Experience Platform Mobile SDKs as part of the Mobile Core extension. These services provide shared functionality throughout the SDK that can be shared by extensions. For example, services provide shared functionality for networking, data queuing, and caching. For more information on services provided by the SDK please see documentation in our [iOS](https://github.com/adobe/aepsdk-core-ios/blob/main/Documentation/Services/README.md) and [Android](https://github.com/adobe/aepsdk-core-ios/tree/main/Documentation/Services) repositories.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/platform-services/index.md:5]: "... please see the documentation in our..."

File: src/pages/documentation/mobile-core/platform-services/index.md [hunk 5]
```diff
@@ -2,23 +2,15 @@ import Tabs from './tabs/index.md'
...
 
 ## Accessing services
 
-The MobileCore extension provides a shared `ServiceProvider`, responsible for accessing the current set of provided services.
+The MobileCore extension provides a shared `ServiceProvider`, responsible for maintaining the current set of provided services and any potential service overrides.
+
+Some services provide wrapper classes. For example, the `Log` class is a wrapper around the `LoggingService`. However, in some cases, a wrapper class may not exist, and one might need to access a service directly from the `ServiceProvider`. The recommended way to do this is through a computed variable or directly through the ServiceProvider when required. This ensures that if the service is overridden, the service consumer always uses the correct service implementation.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/platform-services/index.md:11]: "... may not exists, and you may need to access a service directly..."

Comment (calebk1) [reply src/pages/documentation/mobile-core/platform-services/index.md:11]: We also should put that last mention of ServiceProvider in the code `` things.

**Review by calebk1 — COMMENTED** [review]
Just a few more comments!

File: src/pages/documentation/mobile-core/platform-services/tabs/network-service.md [hunk 6]
```diff
@@ -29,4 +48,229 @@ let networkService = ServiceProvider.shared.networkService
...
+
+```java
+public class MyApp extends Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+		// Set the network override prior to making any other calls to the SDK
+		ServiceProvider.getInstance().seNetworkService(new MyCustomNetworkService());
```
Comment (praveek) [inline src/pages/documentation/mobile-core/platform-services/tabs/network-service.md:226]: typo `setNetworkService`

File: src/pages/documentation/mobile-core/platform-services/tabs/network-service.md [hunk 7]
```diff
@@ -1,20 +1,39 @@
...
+import com.adobe.marketing.mobile.services.NetworkCallback;
+import com.adobe.marketing.mobile.services.NetworkRequest;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+
+final int CONNECTION_TIMEOUT_MS = 5000;
+final int READ_TIMEOUT_MS = 5000;
+final String URL_TO_CONNECT = "URL_TO_CONNNECT_TO";
+final byte[] body = "SampleRequestBody".getBytes(StandardCharsets.UTF_8);
+final Map<String, String> headers = new HashMap<>();
+headers.put("myHeaderName", "myHeaderValue");
```
Comment (praveek) [inline src/pages/documentation/mobile-core/platform-services/tabs/network-service.md:15]: nit `headers.put("customHeaderName", "customHeaderValue");`

File: src/pages/documentation/mobile-core/platform-services/tabs/network-service.md [hunk 8]
```diff
@@ -1,20 +1,39 @@
...
+					// handle connection response
+					
+					// close the response
+					if (response != null) {
+						response.close();
+					}
+				};
+
+// Make the request
+ServiceProvider.getInstance().getNetworkService().connectAsync(networkRequest, networkCallback);
```
Comment (praveek) [inline src/pages/documentation/mobile-core/platform-services/tabs/network-service.md:31]: Can you make the lambda inline and remove `NetworkCallback` import?

**Review by praveek — APPROVED** [review]

**Review by calebk1 — APPROVED** [review]
LGTM!

**Pull request merged**